R/seroprevalence_data.R

```diff
@@ -120,4 +120,131 @@ prepare_bin_data <- function(serodata) {
+                              test = "fake",
+                              antibody = "IgG",
+                              seed = 1234
+                              ){
+    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+        mutate(tsur = tsur,
+            country = 'None',
+            test = test,
+            antibody = antibody,
+            survey = survey_label,
+            age_mean_f = tsur - birth_year)
+    sim_data <- sim_data %>%
+        mutate(counts = get_sim_counts(sim_data, foi, size_age_class, seed = seed),
+            total = size_age_class) %>%
+        prepare_serodata(add_age_mean_f = FALSE)
+
+    return(sim_data)
+}
+
+# TODO: Complete the documentation of group_sim_data
+#' Function that generates  grouped simulated data from a given Force-of-Infection
+#'
+#' @param sim_data Dataframe with the structure of the output of \code{\linl{generate_sim_data}}.
+#' @return Dataframe object containing grouped simulated data generated from \code{foi}
+#' @examples
+#'\dontrun{
+#' size_age_class = 5
+#' foi <- rep(0.02, 50)
+#' sim_data <- generate_sim_data(foi = foi,
+#'                               size_age_class = size_age_class,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' sim_data_grouped <- group_sim_data(sim_data = sim_data,
+#'                                    foi = foi,
+#'                                    size_age_class = size_age_class,
+#'                                    tsur = 2050,
+#'                                    birth_year_min = 2000,
+#'                                    survey_label = 'sim_constant_foi_grouped')
+#' }
+#' @export
+group_sim_data <- function(sim_data,
+                          foi,
+                          size_age_class,
+                          tsur,
+                          birth_year_min,
+                          survey_label,
+                          test = "fake",
+                          antibody = "IgG",
+                          seed = 1234) {
+
+    sim_data <- sim_data %>% mutate(age_group = 'NA', age = age_mean_f) %>% arrange(age)
+    sim_data$age_group[sim_data$age > 0 & sim_data$age < 5] <-   '01-04'
+    sim_data$age_group[sim_data$age > 4 & sim_data$age < 10] <-  '05-09'
+    sim_data$age_group[sim_data$age > 9 & sim_data$age < 15] <-  '10-14'
+    sim_data$age_group[sim_data$age > 14 & sim_data$age < 20] <- '15-19'
+    sim_data$age_group[sim_data$age > 19 & sim_data$age < 25] <- '20-24'
+    sim_data$age_group[sim_data$age > 24 & sim_data$age < 30] <- '25-29'
+    sim_data$age_group[sim_data$age > 29 & sim_data$age < 35] <- '30-34'
+    sim_data$age_group[sim_data$age > 34 & sim_data$age < 40] <- '35-39'
+    sim_data$age_group[sim_data$age > 39 & sim_data$age < 45] <- '40-44'
+    sim_data$age_group[sim_data$age > 44 & sim_data$age < 51] <- '45-50'
```

Comment:
This all seems very hard coded. I would have thought that users would have wanted to have control over how their ages were grouped?

Reply:
Addressed by commits [b5daf13](https://github.com/epiverse-trace/serofoi/commit/b5daf13566cb9d893a0a0f4d7f6a54b8b45416d0) and [9e58cdf](https://github.com/epiverse-trace/serofoi/commit/9e58cdf211e67d97b89c6dbb11373a17f83b2e9a). 

What I did here was to borrow the function `get_age_group` from [vaccineff](https://github.com/epiverse-trace/vaccineff), created by @davidsantiagoquevedo, and I used it to generate `sim_data$age_group`. Then, by means of regular expressions, I generated the columns `age_min` and `age_max` that delimit the intervals. 

In commit [9e58cdf](https://github.com/epiverse-trace/serofoi/commit/9e58cdf211e67d97b89c6dbb11373a17f83b2e9a) I added the option `right = FALSE` to `cut` in `get_age_group`, which means that the intervals are of the form [b1, b2) (see _Details_ [here](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/cut)).